internals/webhooks/subnamespace_webhook.go

```diff
@@ -271,6 +275,25 @@ func ValidateCreateSnsRequest(sns *utils.ObjectContext, parentQuotaObj *utils.Ob
 	return true, ""
 }
 
+// ValidateAllResourceQuotaParamsValid validates that all quota params: storage, cpu, memory, gpu exists and are positive
+// if one of the params is not valid, the function returns false with the relevant error
+func ValidateAllResourceQuotaParamsValid(sns *utils.ObjectContext) (bool, error) {
+	quotaSNS := utils.GetSnsQuotaSpec(sns.Object).Hard
+	ResourceQuotaParams := []corev1.ResourceName{"basic.storageclass.storage.k8s.io/requests.storage", "cpu", "memory", "requests.nvidia.com/gpu", "pods"}
```

Comment:
Instead of having the ResourceNames hard-coded here, I would compare against the ZeroedQuota parameters defined in the `types.go` file, since if we ever decide to change the parameters of what is enforced we'd have to change them there and we want a single source of truth.

https://github.com/dana-team/hns/blob/95ab353af35c7dfcc00cdec5e8329b3204732932/api/v1/types.go#L118-L120

Reply:
Also, this check should only be done if it's not a ResourcePool; it would always fail for ResourcePools